ue4cli/UnrealManagerBase.py

```diff
@@ -44,13 +44,13 @@ def setEngineRootOverride(self, rootDir):
 		# Set the new root directory
 		rootDir = os.path.abspath(rootDir)
 		ConfigurationManager.setConfigKey('rootDirOverride', rootDir)
-		print('Set engine root path override: {}'.format(rootDir))
+		Utility.printStderr('Setting engine root path override:', str(rootDir))
 		
 		# Check that the specified directory is valid and warn the user if it is not
 		try:
 			self.getEngineVersion()
 		except:
-			print('Warning: the specified directory does not appear to contain a valid version of the Unreal Engine.')
+			UnrealManagerException('the specified directory does not appear to contain a valid version of the Unreal Engine.')
```

Comment:
Is this missing a `raise`? Also, since we're in an exception handler here, should we pull some details from the currently-happening exception as well and use `raise from None` to suppress the chained exception reporting, or trivially use `raise from e` to tie them together more nicely than "exception happened during exception handling".

Reply:
Nice catch @TBBle !

I indeed missed the `raise` here. Not sure why this place uses `try-except` block though. It was never pulling the details about the exception so I also didn't try to change this.

Sorry, I haven't noticed your code review, as I was using a mobile app.

Reply:
Funny thing worth noting: even with all warnings enabled my LSP (pylsp) does not warn about this.
![image](https://github.com/adamrehn/ue4cli/assets/91839286/4e75b934-9e35-483e-80ad-3a95afc67f18)

Reply:
Fixed with https://github.com/adamrehn/ue4cli/pull/65/commits/c17c247a4f09e9281db24c11e1239ab32b8c012f

Reply:
I think this should be
```python
		except err:
			raise UnrealManagerException('the specified directory does not appear to contain a valid version of the Unreal Engine.') from err
```
so that the exception will stringify as "was caused by" rather than "during exception processing, another exception was raised".

Another option is
```python
		except:
			raise UnrealManagerException('the specified directory does not appear to contain a valid version of the Unreal Engine.') from None
```
which basically discards the underlying exception; I'd lean away from that option myself, particularly since the underlying code could be raising either an error from the JSON library, or another UnrealManagerException, and the latter doesn't match this failure. (This is why I lean towards wrapping all exceptions at the point they enter our domain, so that we don't have to do more conditional wrapping later.)

Reply:
Created another issue about this https://github.com/adamrehn/ue4cli/issues/67
Fixing bare exception is not a purpose of this PR. I might take care of it later.